src/fetchers/tarball-fetcher.js

```diff
@@ -93,12 +93,39 @@ export default class TarballFetcher extends BaseFetcher {
   } {
     const integrityInfo = this._supportedIntegrity();
 
+    const now = new Date();
+
+    const fs = require('fs');
+    const patchedFs = Object.assign({}, fs, {
```

Comment:
Ugh, why are we patching fs where we already have our own fs functions module?

Reply:
Because it's not the same. The `tar-fs` package is doing `utimes` call in order to set the files timestamps. This function is broken and doesn't support subsecond timestamps. The only way to solve this without forking `tar-fs` is to give it an api-compatible version of the `fs` module, patched in order to use `futimes` instead of `utimes`.

Ideally the fix should be done upstream, but this issue is problematic enough that it warrants an hotfix to land faster.

Reply:
Is it possible to patch `tar-fs` rather than `fs`? That seems like a better idea to me.

Reply:
No, utimes is used within [a closure](https://github.com/mafintosh/tar-fs/blob/master/index.js#L213). I also prefer to patch a function with clearly defined (if a bit buggy) inputs/output rather than a random function from a module that can change without notice.

Also note that this code doesn't modify the global fs implementation. It simply creates a new, local, fs instance and pass it to tar-fs using its custom filesystem option, which seems perfect for this use case.

Reply:
I agree this is a good solution in regards to `tar-fs`. It does kind of bloat the file a little. Do you think it'll be possible to move untar-stream outside somehow?

Reply:
Hm maybe it could be moved in `src/utils/fs`, yeah 👍